### ISSUE TITLE
Putting my code in a Microsoft namespace

### ISSUE THREAD

**Issue by ikyriak** [dotnet/AspNetCore.Docs#20535]
[EDIT by guardrex to add the metadata ..... @ikyriak, it's best to open new issues on the repo using the **This page** feedback button at the bottom of the topic, which adds this metadata automatically and pings the author.]

Related Closed Issue: dotnet/AspNetCore.Docs#11435

Can we have an update regarding the suggestion to "Place extension methods in the Microsoft.Extensions.DependencyInjection namespace to encapsulate groups of service registrations." [[REF](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/dependency-injection.md)].

The last comment was on Jun 27, 2019 from @guardrex : "Engineering has looked this topic over multiple times since then and never said to change it. We'll be reviewing updates again when we get to 3.0 updates. I'll ask explicitly in a few months when I work it."

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0d78f540-81fd-1ba9-cb90-591e4ec27d9b
* Version Independent ID: 2251f557-dd3a-ec35-2558-6243337db009
* Content: [Dependency injection in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-5.0)
* Content Source: [aspnetcore/fundamentals/dependency-injection.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/dependency-injection.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
They ask that devs don't **_publish NuGet packages with "Microsoft" in the name of the package._** They sharpen their 🔪😨 when they see that happen! :smile:

@davidfowl ... This guidance was attributed to you by @ardalis when the DI topic was written ...

> Place extension methods in the `Microsoft.Extensions.DependencyInjection` namespace to encapsulate groups of service registrations.

... is that the guidance that you want?

**Comment by guardrex** [comment]
cc: @mkArtakMSFT ... probably can answer this definitively, too.

TL;DR; &mdash; The DI topic states ...

> Place extension methods in the `Microsoft.Extensions.DependencyInjection` namespace to encapsulate groups of service registrations.

... Is the product unit OK :+1: with devs using that "Microsoft"-based namespace for their own apps and libs?

It's just that MS doesn't want devs to use "Microsoft" in NuGet package names. That's been stated several times over the years.

**Comment by Rick-Anderson** [comment]
@JamesNK Can I remove the following suggestion from the doc?

> Place extension methods in the `Microsoft.Extensions.DependencyInjection` namespace to encapsulate groups of service registrations.

**Comment by JamesNK** [comment]
As far as I know that advice is still recommended. Putting DI extension methods in `Microsoft.Extensions.DependencyInjection` means they show up with intellisense.
